### PR TITLE
Add env variable support for number of followable hashtags in feed column

### PR DESCRIPTION
**Pull request by jakkarth**
Pitch
In the same way that we have an environment variable to change the limit on the number of characters for a post, we should have an environment variable to change the limit on the number of hashtags that can be followed in a ui column.

Motivation
Users may want to follow more than four hashtags related to a specific topic. For example, #gardening #permaculture #ecology #urbanecology #urbangardening.

The argument against this in vanilla is a concern about server performance. As instance admins, that should be our call to make. It's up to us to balance the needs of our users vs the capacity of the server we're running on.

See https://github.com/glitch-soc/mastodon/issues/2489

### CODE REVIEW

**Comment by jakkarth** [comment]
@ClearlyClaire @TheEssem This has been sitting for a month. I'd love to be able to include "got a PR accepted for Glitch" in my list of accomplishments for 2023. Any chance one of you could look at it and at least give some feedback?

**Comment by kescherCode** [comment]
(looks like it'll be 2024)

**Comment by jakkarth** [comment]
> (looks like it'll be 2024)

5 merge requests merged in the last 24 hours but I can't even get a reply. 2557 got a reply in 3 days. I was hoping to step up my open-source contributions this year but with maintainers like this it's hard to be interested. Nothing says "we don't want you here" like being ignored.

**Comment by kescherCode** [comment]
I don't think you are actively being ignored. PRs just take a while to get merged here if not discussed before they are made.

**Comment by jakkarth** [comment]
I don't think it's active either, but the result is the same. I had concerns about spending my time investigating this potential feature because there's so little activity around this project. If upstream hadn't already shot down the idea before I came along, I'd have proposed it there. I did open a discussion here #2489 before opening the PR, and it hasn't received any interaction either. I thought the feature was valuable enough to set up a dev environment, learn about the code, figure out how to make the change, and validate it worked. And I wanted to contribute that effort back to the community. Recognizing that interest is a core part of community development for open-source projects; I'm sure the current maintainers are busy with their own feature work and keeping up with upstream, so bringing in additional interested people to help share the load is crucial for the longevity and capacity for the project. Doing that work and not having it so much as acknowledged for 45 days and counting makes it clear that my contributions aren't of interest, and leaves me feeling frustrated and uninterested in continuing to contribute. And I suspect other devs who are considering PRs will have the same process I do of looking at how new contributors are treated to decide if it's worth their time, so it's not just me that gets turned off from contributing. And I think that's a shame. Glitch has a lot of potential, and I'd like to see it grow. I don't have any interest in maintaining my own fork for just my instance, but I probably would have looked for other bugs to fix or features to add here. As it stands, I'm going to leave the PR open in case they see it later, but this isn't a project I'm going to invest my time in.

**Comment by Fl1pp3d0ff** [comment]
It's pretty much going to open a potential security hole to do so, which is
probably why main shot it down...

**Comment by kescherCode** [comment]
@Fl1pp3d0ff A security hole, just from *allowing* an admin to increase the amount of followable hashtags? Doesn't seem plausible to me, as long as the admin doesn't pick 9999 as the value and allows a few users to DoS the instance they're already on.

**Comment by jakkarth** [comment]
> It's pretty much going to open a potential security hole to do so, which is probably why main shot it down...

I don't see how. It's allowing an instance admin to set an environment variable that's used to set the maximum number of hashtags that can be followed in a column. There's no user-exposed configuration here. At worst, the admin could set it to an abnormally large value like 9999 as mentioned by kescherCode, and then a user could subscribe to 9999 hashtags, which could result in a *service disruption.* Just like admins could set the maximum post size to 64gb and a user could upload a post that large.

Upstream didn't want to action the feature request because they believed they knew better what a reasonable tradeoff between convenience (more hashtags) and performance (fewer hashtags) than the instance admins do. It's the exact same scenario as the max character limit, which is specifically why I wanted to migrate to glitch in the first place. It's my instance, let me configure it how I want.

File: .env.production.sample [hunk 1]
```diff
@@ -251,6 +251,9 @@ SMTP_FROM_ADDRESS=notifications@example.com
 # Maximum allowed character count
 MAX_TOOT_CHARS=500
 
+# Maximum allowed hashtags to follow in a feed column
```
Comment (ClearlyClaire) [inline .env.production.sample:254]: I would add a note that it can have significant performance implications

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Hi! Thank you for your contribution, this change looks sound, but I'd advise caution when changing that value, this can result in very heavy database queries.

Therefore, I think the `.env.production.sample` comment should mention that.

File: .env.production.sample (hunk 1 above)
Comment (jakkarth) [reply .env.production.sample:254]: That's a reasonable suggestion, but I'm unclear on exactly how the performance would be impacted. Does going from 4 to 8 double the potential load? quadruple? How does adding a hashtag compare to adding a followed user?

Comment (ClearlyClaire) [reply .env.production.sample:254]: The issue is not so much with the “posts having any of these tags” mode but “posts having all these tags”

Comment (jakkarth) [reply .env.production.sample:254]: Note added.

**Review by ClearlyClaire — APPROVED** [review]

File: .devcontainer/docker-compose.yml [hunk 2]
```diff
@@ -20,6 +20,8 @@ services:
       ES_ENABLED: 'true'
       ES_HOST: es
       ES_PORT: '9200'
+      MAX_TOOT_CHARS: 666
+      MAX_FEED_HASHTAGS: 5
```
Comment (ClearlyClaire) [inline .devcontainer/docker-compose.yml:24]: Better remove these unnecessary changes.

```suggestion
```

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

**Comment by jakkarth** [comment]
Whoops! Looks like I accidentally caught my test change with a `commit -a`. Good catch! Looks like your commit 28d74b8 got that sorted. Anything else you need from me?

**Review by ClearlyClaire — APPROVED** [review]

**Comment by jakkarth** [comment]
It seems not. Best of luck with your project I guess.
